src/metabase/events/schema.clj

```diff
@@ -5,6 +5,20 @@
    [metabase.models.view-log :as view-log]
    [toucan2.core :as t2]))
 
+(defn- with-hydration
+  "Add options to malli schemas to hydrate information when sending system event notifications.
+
+    (events.notification/hydrate! [:map
+                                    [:user_id (with-hydration :user [:model/User :email]) :int]]
```

Comment:
I kind of dislike how this looks, because schema ends up being separated in two pieces (a key and a type) by a hydration information. That was the reason I did `mu/with` (see serialization/api.clj for an example) so that it would look tighter (in a sense that more relevant pieces of data stay together):

```
(-> [:user_id :int]
    (mu/with {:hydrate {:key :user :model [:model/User :email]}}))
```

Maybe `with-hydrate` should be similar?

Reply:
hmm, `[:user_id :int]` is not a valid schema so you can't use it like that.

well we can write something that works with that structure tho, but it'll always assume the input is a schema for key within a map.

Tho I can imagine `with-hydrate` having an `:hydrate-fn` key in the future where it can work with whichever schema type, even with a seq of IDs, so you'd use it like

```clj
[:user_ids {:hydrate-fn (fn [data user-ids] (assoc data :users (t2/select ...)))} [:sequential :int]]
```

That's why I make this function only to modify the options of a schema.

Reply:
yea I was not really happy with this.

Being able to chain this is cool, tho I wouldn't say I like having to type `{:hydrate ...}` every time; I think hiding that structure is a good abstraction; users only need to care about the key and model they need to update.

Please see the updated function and lmk what you think.

Reply:
Yep, I like it more now!